checker/src/utils.rs

```diff
@@ -91,9 +225,15 @@ pub fn summary_key_str(tcx: &TyCtxt<'_, '_, '_>, def_id: DefId) -> String {
         } else {
             name.push('.');
         }
-        name.push_str(component.data.as_interned_str().as_str().get());
+        let component_name = component.data.as_interned_str().as_str();
+        let component_name = component_name.get();
+        if component_name == "{{impl}}" {
```

Comment:
Where does {{impl}} come from now? It looks like you removed the only instance of it? Or will this be used in future commits?

Reply:
The Rust compiler sticks in the {{impl}} when a function is nested inside an impl block. I remove the braces because they are not valid in an identifier. The reason I do that is so that a foreign contract can define something using the mangled name.